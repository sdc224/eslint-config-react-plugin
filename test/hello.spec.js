const { CLIEngine } = require("eslint");

describe("hello", () => {
	it("should report semicolon errors", () => {
		const cli = new CLIEngine({
			useEslintrc: true
		});
		const { results } = cli.executeOnFiles([require.resolve("./hello.js")]);
		expect(
			results.map(result =>
				result.messages.map(message => ({
					...message,
					filePath: ""
				}))
			)
		).toMatchSnapshot();
	});
});
