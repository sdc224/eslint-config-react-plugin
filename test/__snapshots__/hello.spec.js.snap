// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello should report semicolon errors 1`] = `
Array [
  Array [
    Object {
      "column": 1,
      "endColumn": 12,
      "endLine": 2,
      "filePath": "",
      "line": 2,
      "message": "Unexpected console statement.",
      "messageId": "unexpected",
      "nodeType": "MemberExpression",
      "ruleId": "no-console",
      "severity": 1,
    },
    Object {
      "column": 5,
      "endColumn": 6,
      "endLine": 3,
      "filePath": "",
      "line": 3,
      "message": "'b' is assigned a value but never used.",
      "nodeType": "Identifier",
      "ruleId": "no-unused-vars",
      "severity": 2,
    },
    Object {
      "column": 5,
      "endColumn": 6,
      "endLine": 3,
      "filePath": "",
      "fix": Object {
        "range": Array [
          39,
          42,
        ],
        "text": "const",
      },
      "line": 3,
      "message": "'b' is never reassigned. Use 'const' instead.",
      "messageId": "useConst",
      "nodeType": "Identifier",
      "ruleId": "prefer-const",
      "severity": 2,
    },
    Object {
      "column": 1,
      "endColumn": 2,
      "endLine": 4,
      "filePath": "",
      "line": 4,
      "message": "'a' is constant.",
      "messageId": "const",
      "nodeType": "Identifier",
      "ruleId": "no-const-assign",
      "severity": 2,
    },
    Object {
      "column": 1,
      "endColumn": 4,
      "endLine": 4,
      "filePath": "",
      "line": 4,
      "message": "Unary operator '++' used.",
      "nodeType": "UpdateExpression",
      "ruleId": "no-plusplus",
      "severity": 2,
    },
  ],
]
`;
